backup

```diff
@@ -54,10 +66,21 @@ restic --repo="${RESTIC_REPOSITORY}" backup "${backup_args[@]}" "${tag_options[@
 rc=$?
 set -e
 
+success="true"
 if [ $rc -ne 0 ]; then
-  if [ $rc -eq 3 ] && [ -n "${POST_COMMANDS_INCOMPLETE:-}" ]; then
-  	run_commands "${POST_COMMANDS_INCOMPLETE:-}"
+  if [ $rc -eq 3 ]; then
+    if [ "$(check_bool "${SUCCESS_ON_INCOMPLETE_BACKUP:-}")" = "true" ]; then
+      run_commands "${POST_COMMANDS_INCOMPLETE:-}"
+    else
+      success="false"
+      if [ -n "${POST_COMMANDS_INCOMPLETE:-}" ]; then
```

Comment:
Can this `if` statement be removed? You don't perform this check before calling `POST_COMMANDS_INCOMPLETE` above on line 73. I think (and I am not a bash expert) that `run_commands` checks the command exists before running anyway?

Reply:
No, as we explicitely need the `else` branch. This is answered in detail below.